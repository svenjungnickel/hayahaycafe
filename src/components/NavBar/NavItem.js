import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'react-bootstrap';
import { Link } from 'gatsby';
import { navBar__item } from '../../styles/components/NavBar/NavBar.module.scss';

const NavItem = ({ link, name, active }) => (
    <Nav.Item className={navBar__item}>
        <Link
            to={link}
            className={true === active ? 'nav-link active' : 'nav-link'}
            data-cy={`navBarLink${name.replace(/\s/g, '')}`}
        >
            {name}
        </Link>
    </Nav.Item>
);

NavItem.defaultProps = {
    active: false,
};

NavItem.propTypes = {
    link: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    active: PropTypes.bool,
};

export default NavItem;
