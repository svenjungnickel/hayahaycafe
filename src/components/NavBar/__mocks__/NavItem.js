import React from 'react';

const NavItem = () => <nav />;

export default NavItem;
