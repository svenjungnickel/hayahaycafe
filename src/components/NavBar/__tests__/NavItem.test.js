import React from 'react';
import renderer from 'react-test-renderer';
import NavItem from '../NavItem';

describe('NavItem', () => {
    it('renders default (inactive)', () => {
        const link = 'link';
        const name = 'name';

        const component = <NavItem link={link} name={name} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders active', () => {
        const link = 'link';
        const name = 'name with empty space';
        const active = true;

        const component = <NavItem link={link} name={name} active={active} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
