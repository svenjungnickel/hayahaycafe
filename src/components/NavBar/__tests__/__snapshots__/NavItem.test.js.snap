// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem renders active 1`] = `
<div
  className="navBar__item nav-item"
>
  <a
    className="nav-link active"
    data-cy="navBarLinklink"
    href="name"
  >
    link
  </a>
</div>
`;

exports[`NavItem renders default (inactive) 1`] = `
<div
  className="navBar__item nav-item"
>
  <a
    className="nav-link"
    data-cy="navBarLinklink"
    href="name"
  >
    link
  </a>
</div>
`;
